automation/jenkins/aws/manageRepo.sh

```diff
@@ -165,27 +166,22 @@ function push() {
             fi
 
             debug "Pushing the ${REPO_LOG_NAME} repo upstream..."
-            if git push --tags ${REPO_REMOTE} ${REPO_BRANCH}; then
-                # Push succeeded
-                REPO_PUSHED=true
-                break
+            git symbolic-ref -q HEAD
+            RESULT=$? && [[ ${RESULT} -ne 0 ]] && HEAD_DETACHED=true
+            if [[ "${HEAD_DETACHED}" == "false" ]]; then
+                git push --tags ${REPO_REMOTE} ${REPO_BRANCH} && REPO_PUSHED=true && break || \
```

Comment:
```suggestion
                git push ${REPO_REMOTE} ${REPO_BRANCH && git push --tags ${REPO_REMOTE} ${REPO_BRANCH} && REPO_PUSHED=true && break || \
```

Doesn't --tags only push tags, I didn't think it pushed contents as well

Reply:
@roleyfoley, tested it locally - `--tags` option pushes the tags in addition to a specified branch.
So it is pushing only tags if only remote but no branch is specified:
```
git push --tags ${REPO_REMOTE}
```

Still, we can split git push into two commands:
```
git push ${REPO_REMOTE} ${REPO_BRANCH && git push --tags ${REPO_REMOTE} && REPO_PUSHED=true && break || \
```